app/controllers/submissions_controller.rb

```diff
@@ -461,10 +461,19 @@ def view
       value = annotation.value || 0
       line = annotation.line
       problem = annotation.problem ? annotation.problem.name : "General"
-
+      filename = if annotation.position == -1
+                   # position -1 maps to the Autograder Output
+                   "Autograder Output"
+                 elsif @files && annotation.position != 0
+                   # if the submission is an archive, use filename in archive;
+                   # otherwise, use submission filename
+                   Archive.get_nth_filename(@files, annotation.position)
+                 else
+                   @submission.filename
+                 end
```

Comment:
I'm curious as to why do we need a duplication of this code in this controller as well?